vars/neoDeploy.groovy

```diff
@@ -84,13 +84,6 @@ import static com.sap.piper.Prerequisites.checkScript
      * @see dockerExecute
      */
     'dockerOptions',
-    /**
-     * The path to the archive for deployment to SAP CP. If not provided `mtarFilePath` from commom pipeline environment is used instead.
-     */
-    'source'
```

Comment:
[meta] this is not about "remove unnecessary code", but the `source` property is shifted downwards to the parameter keys. In fact there is no code removed, but only re-located.

Since this is basically unrelated to the other change with the deployments I would like to suggest to open another pull request for this. Having unrelated issues (working on the deployments, shifting down parameters) on one pull request makes life harder than it should be. In case we agree on one item (shifting down a parameter) we could in general merge this, but if there are several items bundled together agreed parts have to wait.